### PR TITLE
ID Parsers: compute resources

### PR DESCRIPTION
**Pull request by mbfrahry**
Finishing id parsers for compute resources

### CODE REVIEW

File: internal/services/compute/image_data_source.go [hunk 1]
```diff
@@ -123,38 +124,37 @@ func dataSourceImage() *pluginsdk.Resource {
...
 
 	var img compute.Image
 
 	if !nameRegexOk {
 		var err error
-		if img, err = client.Get(ctx, resGroup, name, ""); err != nil {
+		if img, err = client.Get(ctx, id.ResourceGroup, id.Name, ""); err != nil {
 			if utils.ResponseWasNotFound(img.Response) {
-				return fmt.Errorf("image %q was not found in resource group %q", name, resGroup)
+				return fmt.Errorf("%swas not found", id)
```
Comment (katbyte) [inline internal/services/compute/image_data_source.go:144]:
```suggestion
				return fmt.Errorf("%s was not found", id)
```

**Review by katbyte — APPROVED** [review]
aside from 1 comment LGTM 🧨

File: internal/services/compute/image_data_source.go (hunk 1 above)
Comment (mbfrahry) [reply internal/services/compute/image_data_source.go]: Thank you thank you!

File: internal/services/compute/linux_virtual_machine_resource.go [hunk 2]
```diff
@@ -329,19 +329,19 @@ func resourceLinuxVirtualMachine() *pluginsdk.Resource {
...
-	defer locks.UnlockByName(name, virtualMachineResourceName)
+	locks.ByName(id.Name, virtualMachineResourceName)
+	defer locks.UnlockByName(id.Name, virtualMachineResourceName)
 
-	resp, err := client.Get(ctx, resourceGroup, name, "")
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name, "")
 	if err != nil {
 		if !utils.ResponseWasNotFound(resp.Response) {
-			return fmt.Errorf("checking for existing Linux Virtual Machine %q (Resource Group %q): %+v", name, resourceGroup, err)
+			return fmt.Errorf("checking for existing %s: %+v", id.Name, id.ResourceGroup, err)
```
Comment (mbfrahry) [inline internal/services/compute/linux_virtual_machine_resource.go:344]:
```suggestion
			return fmt.Errorf("checking for existing %s: %+v", id, err)
```

**Pull request merged**